R/derive_param_response.R

```diff
@@ -0,0 +1,248 @@
+#'     filter= PARAMCD == "PD" & AVALC == "Y")`
+#'   would exclude all responses occurring after the first PD date.
+#'
+#'   + For subjects with at least one response before the end of the assessment
+#'   period, `AVALC` is set to `"Y"`, `AVAL` to `1`, and `ADT` to the first
+#'   date when the response occurred.
+#'
+#'   + For all other subjects `AVALC` is set to `"N"`, `AVAL` to `0`, and
+#'    `ADT` to `NA`.
+#'
+#' @param set_values_to Variables to set
+#'
+#'   A named list returned by `vars()` defining the variables to be set for the
+#'   new parameter, e.g. `vars(PARAMCD = "RSP", PARAM = "Response by
+#'   investigator")` is expected.
+#'
+#'   The values must be symbols, character strings, numeric values or `NA`.
+#'
+#' @param subject_keys Variables to uniquely identify a subject
+#'
+#'   A list of symbols created using `vars()` is expected.
+#'
+#' @details
+#'
+#'   1. The Date of the end of the assessment period (e.g. Progressive disease,
+#'    as defined by `pd_source`) is added to the response dataset.
+#'
+#'   1. The response dataset is restricted to observations occurring before
+#'   ** or on ** the date of progressive disease.
+#'
+#'   1. For each subject (with respect to the variables specified for the
+#'   `subject_keys` parameter), the first observation (with respect to
+#'   `ADT`) where the response condition (`filter_source` parameter) is
+#'   fulfilled is selected.
+#'
+#'   1. For each observation in `dataset_adsl` a new observation is created.
+#'   + For subjects with a response `AVALC` is set to `"Y"`, `AVAL` to `1`,
+#'   and `ADT` to  the first date (`ADT`) where the response condition is
+#'   fulfilled.
+#'   + For all other subjects `AVALC` is set to `"N"`, `AVAL` to `0`
+#'   and `ADT` to `NA`.
+#'
+#'   1. The variables specified by the `set_values_to` parameter are added to
+#'   the new observations.
+#'
+#'   1. The new observations are added to input dataset.
+#'
+#' @author Samia Kabi
+#'
+#' @return The input dataset with a new parameter indicating if and when a
+#' response occurred
+#'
+#' @keywords derivation bds
+#'
+#' @export
+#'
+#' @examples
+#' library(dplyr)
+#' library(admiral)
+#' library(lubridate)
+#' library(rlang)
+#'
+#' adsl <- tibble::tribble(
+#'   ~USUBJID,
+#'   "1",
+#'   "2",
+#'   "3",
+#'   "4"
+#' ) %>%
+#'   mutate(STUDYID = "XX1234")
+#'
+#' adrs0 <- tibble::tribble(
+#'   ~USUBJID, ~ADTC, ~AVALC,
+#'   "1", "2020-01-02", "PR",
+#'   "1", "2020-02-01", "CR",
+#'   "1", "2020-03-01", "CR",
+#'   "1", "2020-04-01", "SD",
+#'   "2", "2021-06-15", "SD",
+#'   "2", "2021-07-16", "PD",
+#'   "2", "2021-09-14", "PD",
+#'   "3", "2021-09-14", "SD",
+#'   "3", "2021-10-30", "PD",
+#'   "3", "2021-12-25", "CR"
+#' ) %>%
+#'   mutate(
+#'     STUDYID = "XX1234",
+#'     ADT = ymd(ADTC),
+#'     PARAMCD = "OVR",
+#'     PARAM = "Overall Response",
+#'     ANL01FL = "Y"
+#'   ) %>%
+#'   select(-ADTC)
+#'
+#' # Add a parameter for the date of Progressive disease (PARAMCD = PD)
```

Comment:
i would just include these steps and the above in the tribble data directly (i.e. just give input data with this param already covered) so that the user only needs to see the impact of the derive_param_response step here. they'll see the step by step full instructions of the ADRS workflow in the ADRS vignette